episodes/10-explore-and-plot-by-vector-layer-attributes.Rmd

```diff
@@ -103,79 +101,84 @@ Note that this way the values are shown by default in alphabetical order and `NA
 
 Explore the attributes associated with the `point_Delft` spatial object.
 
-1. How many attributes does it have?
+1. How many fields does it have?
 2. What types of leisure points do the points represent? Give three examples.
-3. Which of the following is NOT an attribute of the `point_Delft` data object?
+3. Which of the following is NOT a field of the `point_Delft` object?
 
-  A) location B) leisure C) osm_id
+  A) `location` B) `leisure` C) `osm_id`
 
 :::::::::::::::::::::::: solution 
 
-1. To find the number of attributes, we use the `ncol()` function:
+1. To find the number of fields, we use the `ncol()` function:
 
-```{r}
+```{r points-ncol}
 ncol(point_Delft)
 ```
 
 2. The types of leisure point are in the column named `leisure`.
 
 Using the `head()` function which displays 6 rows by default, we only see two values and `NA`s. 
 
-```{r}
+```{r points-head}
 head(point_Delft)
 ```
 
-We can increase the number of rows with the n argument (e.g., `head(n = 10)` to show 10 rows) until we see at least three distinct values in the leisure column. Note that printing an `sf` object will also display the first 10 rows.
+We can increase the number of rows with the `n` argument (e.g., `head(n = 10)` to show 10 rows) until we see at least three distinct values in the leisure column. Note that printing an `sf` object will also display the first 10 rows.
 
-```{r}
-head(point_Delft, 10)
+```{r points-head-10}
 # you might be lucky to see three distinct values
+head(point_Delft, 10)  
 ```
 
 We have our answer (`sports_centre` is the third value), but in general this is not a good approach as the first rows might still have many `NA`s and three distinct values might still not be present in the first `n` rows of the data frame. To remove `NA`s, we can use the function `na.omit()` on the leisure column to remove `NA`s completely. Note that we use the `$` operator to examine the content of a single variable.
 
-```{r}
-head(na.omit(point_Delft$leisure)) # this is better
+```{r points-leisure}
+# this is better
+na.omit(point_Delft$leisure) %>% head()  
 ```
 
 To show only unique values, we can use the `levels()` function on a factor to only see the first occurrence of each distinct value. Note `NA`s are dropped in this case and that we get the first three of the unique alphabetically ordered values.
 
-```{r}
-head(levels(factor(point_Delft$leisure)), n = 3)
+```{r points-leisure-head-3}
 # this is even better
+factor(point_Delft$leisure) %>%
+  levels() %>%
+  head(n = 3)   
 ```
 
-3. To see a list of all attribute names, we can use the `names()` function.
+3. To see a list of all attribute names and answer the last question, we can use the `names()` function.
```

Comment:
When first presenting the question, "attribute" was changed to  "field".  Proposed change to keep consistency:
```suggestion
3. To see a list of all fields names and answer the last question, we can use the `names()` function.
```